mitodl/mitxonline
# Adds a feature flag (overridable by URL) for the program UI

#### Pre-Flight checklist


#### What are the relevant tickets?

Fixes #915

#### What's this PR do?

Puts the program UI on the dashboard behind a feature flag. If the program UI is disabled, it can be enabled by appending `?enable_programs` to the URL. 

#### How should this be manually tested?

You will need enrollments in a program to test.

1. Add `FEATURE_ENABLE_PROGRAM_UI=False` to your `.env`.
2. Rebuild and load the app. Navigate to the Dashboard.
3. Course enrollments should display as they did before the program functionality was added. (In other words, you should see a course card for each enrollment and there shouldn't be any program specific cards.) 
4. Either set `FEATURE_ENABLE_PROGRAM_UI` to `True` in the `.env` or append `?enable_programs` to the URL for the dashboard.
5. You should now see program cards on the dashboard.

#### Screenshots (if appropriate)

Note that these screenshots are prior to https://github.com/mitodl/mitxonline/pull/926 so they don't include the updated dashboard UI. 

With the UI disabled:
![Screen Shot 2022-09-01 at 3 18 18 PM](https://user-images.githubusercontent.com/945611/188007120-b6b36b0a-d541-4861-b804-6dc30ce8f354.png)

With the UI enabled (using the URL param):

![Screen Shot 2022-09-01 at 3 26 12 PM](https://user-images.githubusercontent.com/945611/188007176-7e06e0c1-0480-47ce-9efc-0c1f10496f97.png)
